science/paraview/Portfile

```diff
@@ -56,6 +57,7 @@ depends_lib-append  path:include/eigen3/Eigen/Eigen:eigen3 \
                     port:jpeg \
                     port:libpng \
                     port:tiff \
+                    port:icet \
```

Comment:
Is this still required, given you are enforcing the icet mpi variant anyway ? Doesn’t that variant add the dep itself in which case it is not required here ?

Reply:
Mmmh, this is the procedure
```
proc mpi.action_enforce_variants {ports} {
    global name
    ui_debug "mpi.enforce_variant list: ${ports}"
    foreach depspec $ports {
        set portname [_get_dep_port $depspec]
        if {![catch {set result [active_variants $portname "" ""]}]} {
            set otmpi  [mpi_active_variant_name $portname]
            set mympi  [mpi_variant_name]

            if {$otmpi ne "" && $mympi eq ""} {
                # instead of trying to append to a default variant (which fails
                # silently for some cases), we should be explicit and tell the
                # user about the error
                ui_error "Need to select variant +$otmpi"
                return -code error "$portname +$otmpi is installed, so +$otmpi needs to be selected for $name"
            } elseif {$otmpi ne $mympi} {
                ui_error "Install $portname +$mympi"
                return -code error "$portname +$mympi not installed"
            }
        } else {
            ui_error "Internal error: '$portname' does not refer to an installed port."
        }
    }
}
```
It doesn't look like it adds the port in the deps... Am I wrong?

Reply:
Hmm. Does the mpi PG even know about ice ? I see no mention of it in

https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/mpi-1.0.tcl

?? Does

```
mpi.enforce_variant icet
```

actually do anything ?

Reply:
What it does, looking into the procedure I pasted, is just warning the user to install the mpi variant of that dependency if she doens't. Nothing more.

Reply:
This is my understanding as well. LGTM.